Coin selector

AlexeyTripleA opened the pull request.

pshenmic committed:
Upgrade CI to node 16

derrickhorton committed:
Merge pull request #104 from anypay/fix/ci-node-16

Upgrade CI to node 16

AlexeyTripleA committed:
Coin selector created

AlexeyTripleA committed:
Upload package-lock.json, package-lock.json, yarn.lock

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
 
   }, [provider])
 
+  const coinsList:any = {
```
`supportedCoins`

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+  const coinsList:any = {
+    'AVAX': { // Avalanche
+      wallets: ['BRAVE_WALLET'],
+      'isActive': false
```
`enabled`

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+      "title": "RELAYX", 
+      'description': 'Swipe to pay using Relay wallet',
+      'isActive': true,
+      'paymentBlock': <PaymentRelayService
```
Better drop down into the render function

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+
+  type PaymentWalletArray = {[key: string]: PaymentWallet;};
+
+  const walletsArry: PaymentWalletArray = {
```
`supportedWallets`

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+      }
+    });
+
+    selectorOptions.push({value: coinTitle, label: coinTitle, isDisabled: isDisabled});
```
Can be a type, so using any could be avoided

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+
+  const selectorOptions:any = [];
+
+  for (var coinTitle in coinsList) {
```
Use filter + map

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,6 +599,206 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+  const [selectedCoin, setSelectedCoin] = useState("BSV");
+
+
+  function coinSelectorChangeHandler (selectedOption: any) {
```
`onSelectorChange`

pshenmic reviewed (changes requested):
Very good start

AlexeyTripleA committed:
coin selector upd:
+ added types for objects
+ refactoring code of sort cycles
+ change variables and functions names

AlexeyTripleA committed:
coin selector upd:
+ Create payment buttons
+ Upd list filter
+ Added Dash payment ui
+ Added Brave Wallet payment ui

AlexeyTripleA committed:
add borders for payment buttons

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+    'wallets': string[];
+  };
+  
+  type SupportedCoinsArray = {[key: string]: CoinInfo;};
```
Lets move types in the top

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+    'LTC': { // Litecoin
+      wallets: ['EDGE'],
+    },
+    'MATIC': { // Polygon
```
Remove quotes

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+    })
+);
+
+  const [selectedCoin, setSelectedCoin] = useState(selectorOptions[0].value);
```
Use destruction `const [selectorOption] = selectorOptions`

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+
+  const [selectedCoin, setSelectedCoin] = useState(selectorOptions[0].value);
+
+  function onCoinSelectorChange (selectedOption: any) {
```
To the top as well

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+        return <></>
+      }
+
+      let onClick = paymentWallet.handler;
```
Inline variable

pshenmic commented on src/components/Payments/PaymentsOptions/index.tsx:
```diff
@@ -596,16 +684,327 @@ function PaymentsOptionsComponent({ paymentOptions }: any) {
+    
+    <div style={{display: 'none'}}>
+
+    {/* old list */}
```
Lets remove old list

AlexeyTripleA committed:
remove quotation mark from objects titles

AlexeyTripleA committed:
Move constants to the top in code,
remove unusable imports

AlexeyTripleA committed:
remove unusable constant and comments

AlexeyTripleA committed:
Merge remote-tracking branch 'origin/develop' into coin_selector
erge develop into coin_selector

AlexeyTripleA committed:
added keys attributes for payment buttons

AlexeyTripleA committed:
remove caniuse-lite from package.json

AlexeyTripleA committed:
Fix bug with unknown coin in payment options

AlexeyTripleA committed:
test env variable setting

AlexeyTripleA committed:
set baseURL from env variable
